src/cpp/include/openvino/genai/generation_config.hpp

```diff
@@ -71,6 +73,10 @@ class OPENVINO_GENAI_EXPORTS GenerationConfig {
     size_t max_length = SIZE_MAX;
     bool ignore_eos = false;
     size_t min_new_tokens = 0;
+    
+    // OpenAI exposes stop as an array of strings, but vLLM allows also specifying tokens so we expose both.
+    std::vector<std::string> stop_strings;
+    std::vector<std::vector<int64_t>> stop_token_ids;
```

Comment:
can you ensure that for cases of multiple EOS tokens like here https://huggingface.co/microsoft/Phi-3-mini-128k-instruct/blob/main/generation_config.json#L4, you read them to `stop_token_ids` ?

Looks like we need to think how to synchronize this new field and eos_token_id, because they can be set to different values and it's confusing

Reply:
I'm not sure about that. I think `stop_token_ids` should be treated more like `stop_strings` than `eos_token_id`, so to simply cut generation. If we merge multiple eos token ids to `stop_token_ids` how are `ignore_eos` option and eos penalty going to work? Perhaps multiple eos token handling should be implemented separately?

Reply:
IMO, multiple EOS tokens is the same `stop_token_ids` when inner vector has a single element, is not it?

Reply:
A model usually have multiple `eos_token_id`. It's just a GenAI limitation. I'd expect `stop_token_ids` and `eos_token_id` to be the same thing.

> how are `ignore_eos` option and eos penalty going to work?

It's worth checking with the project you are trying to replicate. The chances are that every `stop_token_id` contributes to penalty.